src/crimson/mon/MonClient.cc

```diff
@@ -1024,14 +1036,32 @@ Client::run_command(const std::vector<std::string>& cmd,
   m->cmd = cmd;
   m->set_data(bl);
   auto& req = mon_commands[tid];
-  return active_con->get_conn()->send(m).then([&req] {
+  return send_message(m).then([&req] {
     return req.get_future();
   });
 }
 
 seastar::future<> Client::send_message(MessageRef m)
 {
-  return active_con->get_conn()->send(m);
+  if (active_con) {
+    if (pending_messages.size()) {
+      send_pendings();
+    }
+    return active_con->get_conn()->send(m);
+  }
+  pending_messages.emplace_back(m);
+  return pending_messages.back().pr.get_future();;
```

Comment:
i'd suggest move this block into an `else` clause for better readability.

Reply:
```suggestion
 auto delayed = pending_messages.emplace_back(m);
 return delayed->pr.get_future();;
```

Reply:
This seems not feasible, seastar::promise can't be copied.

Reply:
```suggestion
 auto& delayed = pending_messages.emplace_back(m);
 return delayed->pr.get_future();;
```

@xxhdx1985126 how about chaning `auto` to `auto&` as above?